lib/wifi_user/use_case/sns_notification_handler.rb

```diff
@@ -56,4 +60,17 @@ def handle_sponsor_request(payload)
 
     sponsor_signup_handler.execute(sponsee_extractor.execute, from_address)
   end
+
+  def request_invalid?(request)
+    !request_valid?(request)
+  end
+
+  def request_valid?(request)
+    # For now, we only care that the correct header is set to see if we're
+    # actually dealing with a notification.
+    # There is much more that should be in here.
+
+    request.has_header?('x-amz-sns-message-type') \
+    && request.get_header('x-amz-sns-message-type') == 'Notification'
```

Comment:
Was aiming to push the knowledge of the internals of this request down into the parser object I created.
Feels like we need a domain object that has a known interface that we can return to this use case?

Happy to discuss.